test/unix-socket.ts

```diff
@@ -68,4 +68,32 @@ if (process.platform !== 'win32') {
 		const url = format('http://unix:%s:%s', server.socketPath, '/');
 		t.is((await got(url)).body, 'ok');
 	});
+
+	test('`unix:` fails when UNIX sockets are not enabled', async t => {
+		const gotUnixSocketsDisabled = got.extend({enableUnixSockets: false});
+
+		t.false(gotUnixSocketsDisabled.defaults.options.enableUnixSockets);
+		await t.throwsAsync(
+			gotUnixSocketsDisabled('unix:'),
+			{
+				code: 'ERR_UNSUPPORTED_PROTOCOL',
+			},
+		);
+	});
+
+	test('`http://unix:/` fails when UNIX sockets are not enabled', async t => {
+		const gotUnixSocketsDisabled = got.extend({enableUnixSockets: false});
+
+		t.false(gotUnixSocketsDisabled.defaults.options.enableUnixSockets);
+
+		try {
+			await got('http://unix:');
+		} catch (error: any) {
+			t.regex(error.code, /ENOTFOUND|EAI_AGAIN/);
```

Comment:
We want to check with regex here. Seems `t.throwsAsync` has not supported regex in `code` yet.

Reply:
Why `EAI_AGAIN`?

Reply:
`t.throwsAsync` returns the error too. So you can check the return value.

Reply:
`EAI_AGAIN` is for ubuntu. The error code there is different from mac & win.

I log the error in a previous [test](https://github.com/sindresorhus/got/runs/6949849768?check_suite_focus=true#step:5:729).
![Screenshot 2022-07-09 at 1 02 18 PM](https://user-images.githubusercontent.com/46992350/178092363-ff3829e1-4159-4946-a615-63c2b714f837.png)

Reply:
This seems incorrect. The error codes mean that the request actually went through (even though it wasn't unix, it looked up the `unix` hostname instead).